### PR TITLE
Fixes issue 598.

### PR DESCRIPTION
**Pull request by lemire**
Adopts the new API.

### CODE REVIEW

File: tools/json2json.cpp [hunk 1]
```diff
@@ -72,33 +72,18 @@ int main(int argc, char *argv[]) {
...
     return EXIT_FAILURE;
   }
   if (apidump) {
-    simdjson::ParsedJson::Iterator pjh(pj.doc);
-    if (!pjh.is_ok()) {
-      std::cerr << " Could not iterate parsed result. " << std::endl;
-      return EXIT_FAILURE;
-    }
-    compute_dump(pjh);
+    simdjson::document::iterator i(doc);
```
Comment (jkeiser) [inline tools/json2json.cpp]: Can we replace this with `cout << doc` or is compute_dump doing something different than that?

Comment (jkeiser) [reply tools/json2json.cpp]: Trying to discourage the use of document::iterator and ParsedJson::Iterator both now that we have the DOM API. Actually, I think we need to remove document::iterator.

Comment (lemire) [reply tools/json2json.cpp]: Point taken. I removed this whole thing entirely. The program is now much leaner... which is the point... our own tools should be good citizens.

benchmark/bench_dom_api.cpp has references to document::iterator though maybe it is intentional.

**Comment by lemire** [comment]
I'll merge when all checks pass.

**Pull request merged**